source/mesh_deformation/free_surface.cc

```diff
@@ -307,15 +307,27 @@ namespace aspect
 
                 cell_vector = 0;
                 cell_matrix = 0;
+                Tensor<1,dim> direction;
+                if (advection_direction == SurfaceAdvection::geometric_normal)
+                  // Only for 2D
+                  {
+                    Tensor<1,dim> tangent = (fs_fe_face_values.quadrature_point(n_face_q_points-1) - fs_fe_face_values.quadrature_point(0));
+                    direction[0] = -tangent[1];
+                    direction[1] = tangent[0];
+                    // Check that normal points upwards
+                    if (direction[1] < 0)
+                      direction *= -1.;
```

Comment:
This seems very specific to the box geometry. If that's what you want, you should probably also add an assertion for that.